prov/tcp/src/tcpx_progress.c

```diff
@@ -151,28 +177,46 @@ static int tcpx_process_remote_write(struct tcpx_pe_entry *pe_entry)
 		pe_entry->msg_data.iov[i].iov_len =
 			pe_entry->msg_hdr.rma_iov[i].len;
 	}
-	return FI_SUCCESS;
 }
 
-static int tcpx_validate_rx_rma_data(struct tcpx_pe_entry *pe_entry)
+static void tcpx_prepare_remote_read_resp(struct tcpx_pe_entry *resp_entry)
+{
+	int i;
+
+	resp_entry->msg_data.iov[0].iov_base = (void *) &resp_entry->msg_hdr;
+	resp_entry->msg_data.iov[0].iov_len = sizeof(resp_entry->msg_hdr);
+	resp_entry->msg_data.iov_cnt = 1 + resp_entry->msg_hdr.rma_iov_cnt;
+
+	resp_entry->msg_hdr.hdr.size = resp_entry->msg_data.iov[0].iov_len;
+	for ( i = 0 ; i < resp_entry->msg_hdr.rma_iov_cnt ; i++ ) {
+		resp_entry->msg_data.iov[i+1].iov_base =
+			(void *) resp_entry->msg_hdr.rma_iov[i].addr;
```

Comment:
nit: I think the cast here needs to be (void *) (uintptr_t) to handle 32-bit architectures.